libcxx/test/std/experimental/simd/simd.nonmembers/simd_reductions.pass.cpp

```diff
@@ -0,0 +1,221 @@
+//                                 bit_and<> binary_op) noexcept;
+// template <class M, class V>
+//   typename V::value_type reduce(const const_where_expression<M, V>& x,
+//                                 bit_or<> binary_op) noexcept;
+// template <class M, class V>
+//   typename V::value_type reduce(const const_where_expression<M, V>& x,
+//                                 bit_xor<> binary_op) noexcept;
+
+// template <class T, class Abi>
+//   T hmin(const simd<T, Abi>&) noexcept;
+// template <class M, class V>
+//   typename V::value_type hmin(const const_where_expression<M, V>&) noexcept;
+// template <class T, class Abi>
+//   T hmax(const simd<T, Abi>&) noexcept;
+// template <class M, class V>
+//   typename V::value_type hmax(const const_where_expression<M, V>&) noexcept;
+
+#include "../test_utils.h"
+#include <cassert>
+#include <experimental/simd>
+
+namespace ex = std::experimental::parallelism_v2;
+#include <iostream>
+struct CheckSimdReduction {
+  template <class _Tp, class SimdAbi>
+  void operator()() {
+    const ex::simd<_Tp, SimdAbi> lhs(static_cast<_Tp>(5));
+    const ex::simd<_Tp, SimdAbi> rhs([](_Tp i) { return i; });
+
+    if constexpr (!std::is_floating_point_v<_Tp>) // ex::reduce has not finish?
+    {
+      const ex::simd<_Tp, SimdAbi> simd_([](_Tp i) { return i; });
+      _Tp result = ex::reduce(simd_, std::plus<>());
+      _Tp expected{};
+      for (size_t i = 0; i < simd_.size(); ++i)
+        expected += simd_[i];
+      assert(result == expected);
+    }
+    if constexpr (!std::is_floating_point_v<_Tp>) // ex::reduce has not finish?
+    {
+      const ex::simd<_Tp, SimdAbi> simd_([](_Tp i) { return i; });
+      _Tp identity_element{};
+      auto reduce_result = ex::reduce(ex::where(simd_ > 0, simd_), identity_element, std::plus<>());
+
+      if (ex::none_of(simd_ > 0)) {
+        assert(identity_element == reduce_result);
+      } else {
+        _Tp result = ex::reduce(simd_, std::plus<>());
+        _Tp expected{};
+        for (size_t i = 0; i < simd_.size(); ++i)
+          expected += simd_[i];
+        assert(result == expected);
+      }
+    }
+    if constexpr (!std::is_floating_point_v<_Tp>) // ex::reduce has not finish?
+    {
+      const ex::simd<_Tp, SimdAbi> simd_([](_Tp i) { return i; });
+      auto reduce_result = ex::reduce(ex::where(simd_ > 0, simd_), std::plus<>());
+
+      if (ex::none_of(simd_ > 0)) {
+        assert(0 == reduce_result);
+      } else {
+        _Tp result = ex::reduce(simd_, std::plus<>());
+        _Tp expected{};
+        for (size_t i = 0; i < simd_.size(); ++i)
+          expected += simd_[i];
+        assert(result == expected);
+      }
+    }
+    if constexpr (!std::is_floating_point_v<_Tp>) // ex::reduce has not finish?
+    {
+      const ex::simd<_Tp, SimdAbi> simd_([](_Tp i) { return i; });
+      auto reduce_result = ex::reduce(ex::where(simd_ > 0, simd_), std::multiplies<>());
```

Comment:
Missing two types of reduce function check for multiplies, bit_and, bit_or, bit_xor.